content_sync/utils.py

```diff
@@ -140,6 +142,27 @@ def get_ocw_studio_api_url():
     return "http://10.1.0.102:8043" if is_dev() else settings.SITE_BASE_URL
 
 
+def get_publishable_sites(version: str):
+    publish_date_field = (
+        "publish_date" if version == VERSION_LIVE else "draft_publish_date"
+    )
+    # Get all sites, minus any sites that have never been successfully published
+    sites = Website.objects.exclude(
+        Q(**{f"{publish_date_field}__isnull": True}) | Q(url_path__isnull=True)
+    )
+    if version == VERSION_LIVE:
+        sites = sites.exclude(unpublish_status__isnull=False)
```

Comment:
This exclusion should apply to both draft and live sites, otherwise the draft mass build pipeline hangs when trying to retrieving the deleted git repo for unpublished sites (locally at least).  If an unpublished site is ever republished, the mass build pipeline should be updated to include it again